cpp/oneapi/dal/algo/basic_statistics/backend/cpu/apply_weights_cpu.cpp

```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+* Copyright 2023 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <algorithm>
+
+#include "oneapi/dal/algo/basic_statistics/backend/cpu/apply_weights.hpp"
+#include "oneapi/dal/detail/threading.hpp"
+#include "oneapi/dal/backend/common.hpp"
+
+namespace oneapi::dal::basic_statistics::backend {
+
+template <typename Cpu, typename Float>
+std::int64_t propose_threading_block_size(std::int64_t row_count, std::int64_t col_count) {
+    using idx_t = std::int64_t;
+    ONEDAL_ASSERT(row_count > 0);
+    ONEDAL_ASSERT(col_count > 0);
+    constexpr idx_t max_block_mem_size = 512 * 1024;
+    const idx_t block = max_block_mem_size / (col_count * sizeof(Float));
```

Comment:
Confusing naming: I can't obviously understand does `block` mean number of rows per block (because mem_size divided by columns*size(dtype)) or else.

Reply:
fixed